src/SWP/Bundle/TemplatesSystemBundle/Provider/WidgetProvider.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Superdesk Web Publisher Template Engine Bundle.
+ *
+ * Copyright 2015 Sourcefabric z.u. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú
+ * @license http://www.superdesk.org/license
+ */
+
+namespace SWP\Bundle\TemplatesSystemBundle\Provider;
+
+use Doctrine\ORM\Query;
+use SWP\Bundle\TemplatesSystemBundle\Repository\WidgetModelRepository;
+
+class WidgetProvider implements WidgetProviderInterface
+{
+    /**
+     * @var WidgetModelRepository
+     */
+    protected $widgetRepository;
+
+    /**
+     * WidgetProvider constructor.
+     *
+     * @param WidgetModelRepository $widgetRepository
+     */
+    public function __construct(WidgetModelRepository $widgetRepository)
+    {
+        $this->widgetRepository = $widgetRepository;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getQueryForAll(): Query
+    {
+        return $this->widgetRepository->getAll();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getOneById(int $id)
```

Comment:
Needed as it gives option to override it (we will do this when widgets will be revisions aware) - then we just need to create new provider and still use this same repository. Provider allows us to modify base query builder for current needs and return final object.